docs/content/staking/epoch-preparation.mdx

```diff
@@ -16,28 +16,143 @@ description: Technical Overview of the Flow Epoch Protocol
+
+The smart contract that determines the nodes for the next Epoch has special privileges.
+Specifically, it is allowed to emit [Service Events](/staking/epoch-scripts-events/#monitor-epoch-service-events),
+which are how the execution state updates the consensus node-based protocol state.
+
+At the end of the staking auction, the epoch smart contracts conclude that they have now determined
+the set of nodes which will be running the network for the next Epoch, and the amount of FLOW that all the nodes have staked.
+The smart contract then emits a service event with this information.
+
+When processing the block with seat assignment, all network nodes (including future ones which are supposed to monitor the chain in anticipation)
+are thereby informed about the upcoming change.
+
+<Callout type="warning">
+Note: At this point in the epoch (end of the staking auction), 
+there is no change in participating nodes.
+The change in participating nodes happens at the end of the epoch.
+</Callout>
+
+After the staking auction, there is an interim period of time until the new Epoch starts for the following tasks to be completed:
+- The epoch smart contract runs the cluster assignment algorithm for all the collector nodes
+and each collector node will vote for the root block of their respective clusters
+- The Random Beacon Committee for the next Epoch (currently all consensus nodes)
+will run the Distributed Key Generation (DKG),
+- When completing the QC generation and DKG, the smart contracts will emit a service event. 
+After consensus nodes have collected all relevant information (DKG Keys), 
+they can update the identity table to include the information for the next Epoch.
+
+If some necessary tasks for the next Epoch are not completed in the interim period of time, Flow halts!
+
+## Epoch Length
+
+The length of an Epoch is measured in terms of consensus views. The number of views in an epoch
+and in the various epoch phases are known before the Epoch begins
+and stored as a field in the main epoch smart contract (`FlowEpoch`).
+
+The length of an epoch *is not* fork-dependent, as it is measured in views.
+Generally, there is not a block for every view.
+However this does not pose a problem, as for each fork,
+the first and last block of each Epoch are clearly defined. 
+
+<Callout type="warning">
+  **Motivation for measuring epoch length in consensus views:**
+
+
+  While we utilize a consensus-internal detail for measuring time,
+  this solution has significantly less engineering complexity compared to using height.
+  The reason is that HotStuff requires (for liveness) a deterministic *fork-independent* primary selection per view.
+  Defining Epoch length based on height would break fork-independent primary selection,
+  as there can be multiple blocks at the same height, but with different views.
+  Using view numbers, there are no extra edge cases. 
+</Callout>
```

Comment:
I feel like this might be too much detail for these docs. To me the important thing to be aware of is that epochs are measured in views, not heights, and that epoch length is defined before the epoch starts.

Reply:
That is a good point. I'll trim this dowm